core/src/test/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCOTest.scala

```diff
@@ -270,23 +274,99 @@ class RecordInsightsLOCOTest extends FlatSpec with TestSparkContext {
     // Strengths of features "A", "B", and "C" should be much larger the other feature strengths
     assert(abcAvg > 4 * otherAvg,
       "Average feature strengths for features involved in label formula should be " +
-      "much larger than the average feature strengths of other features")
+        "much larger than the average feature strengths of other features")
     // There should be a really large t-value when comparing the two avg feature strengths
-    assert(math.abs(abcAvg - otherAvg) / math.sqrt((abcVar + otherVar)/numRows) > 10,
+    assert(math.abs(abcAvg - otherAvg) / math.sqrt((abcVar + otherVar) / numRows) > 10,
       "The t-value comparing the average feature strengths between important and other features should be large")
 
     // Record insights averaged across all records should be similar to the feature importances from Spark's RF
     val rfImportances = sparkModel.getSparkMlStage().get.featureImportances
     val abcAvgRF = abcIndices.map(rfImportances.apply).sum / abcIndices.size
     val otherAvgRF = otherIndices.map(rfImportances.apply).sum / otherIndices.size
-    val avgRecordInsightRatio = math.abs(abcAvg/otherAvg)
-    val featureImportanceRatio = math.abs(abcAvgRF/otherAvgRF)
+    val avgRecordInsightRatio = math.abs(abcAvg / otherAvg)
+    val featureImportanceRatio = math.abs(abcAvgRF / otherAvgRF)
 
     // Compare the ratio of importances between "important" and "other" features in both paradigms
-    assert(math.abs(avgRecordInsightRatio - featureImportanceRatio)*2 /
+    assert(math.abs(avgRecordInsightRatio - featureImportanceRatio) * 2 /
       (avgRecordInsightRatio + featureImportanceRatio) < 0.8,
       "The ratio of feature strengths between important and other features should be similar to the ratio of " +
         "feature importances from Spark's RandomForest")
   }
 
+  it should "aggregate text derived features" in {
```

Comment:
@michaelweilsalesforce @Jauntbox do these test have to be that verbose & explicit? such large tests seem barely readable to comprehend.

Reply:
Some of the verbose come from the creation of the dataset. Not sure we need to move this part at the beginning of the class.

Reply:
When lookin on a test - it has to be easy to understand what are the inputs, outcomes and assertions that are verified in the test; everything else have to elsewhere.